src/main/java/org/spectrumauctions/sats/core/bidlang/generic/SizeOrderedPowerset/GenericSetsPickN.java

```diff
@@ -50,6 +45,23 @@ public GenericSetsPickN(Map<T, Integer> maxQuantities, int target) {
         next();
     }
 
+    /**
+     * Appends 0 to a list until it has a certain size
+     */
+    private static void appendZeros(int size, List<Integer> list) {
+        while (list.size() < size) {
+            list.add(0);
+        }
+    }
+
+    private static int sum(List<Integer> summands) {
+        int result = 0;
```

Comment:
`summands.stream().mapToInt(Integer::intValue).sum()`  
Not too sure if the syntax is right, but streams might be nice here. Not that it matters, though ;-)

Reply:
Of course it does matter! 😀 Good point, didn't really touch this class, so it escaped my eyes.